.github/workflows/cross-compile.yaml

```diff
@@ -0,0 +1,312 @@
+name: cross-compile
+on:
+  workflow_dispatch:
+  push:
+    branches-ignore:
+      - '**/dev2'
+      - '**/*dev2'
+  pull_request:
+
+concurrency: ci-cross-compile-${{ github.ref }}
+
+env:
+  # string with name of libraries to be built
+  BUILD_LIBS: "SuiteSparse_config:Mongoose:AMD:BTF:CAMD:CCOLAMD:COLAMD:CHOLMOD:CSparse:CXSparse:LDL:KLU:UMFPACK:ParU:RBio:SPQR:SPEX:GraphBLAS:LAGraph"
+  # string with name of libraries to be checked
+  CHECK_LIBS: "SuiteSparse_config:Mongoose:AMD:BTF:CAMD:CCOLAMD:COLAMD:CHOLMOD:CSparse:CXSparse:LDL:KLU:UMFPACK:ParU:RBio:SPQR:SPEX:GraphBLAS:LAGraph"
+  # string with name of libraries that are installed
+  INSTALLED_LIBS: "SuiteSparse_config:Mongoose:AMD:BTF:CAMD:CCOLAMD:COLAMD:CHOLMOD:CXSparse:LDL:KLU:UMFPACK:ParU:RBio:SPQR:SPEX:GraphBLAS:LAGraph"
+
+
+jobs:
+
+  cross-compile:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        # Use bash shell as default
+        shell: bash
+
+    strategy:
+      # Allow other runners in the matrix to continue if some fail
+      fail-fast: false
+
+      matrix:
+        arch: [armhf, ppc64el, s390x, riscv64]
+        include:
+          - arch: armhf
+            target-triple: arm-linux-gnueabihf
+            # cross-toolchain: g++-9-arm-linux-gnueabihf
+            # fortran-cross-toolchain: gfortran-14-arm-linux-gnueabihf
+            # cc: "arm-linux-gnueabihf-gcc-9"
+            # cxx: "arm-linux-gnueabihf-g++-9"
+            # fc: "arm-linux-gnueabihf-gfortran-14"
+            ccache-max: 64M
+          - arch: ppc64el
+            target-triple: powerpc64le-linux-gnu
+            # cross-toolchain: g++-9-powerpc64le-linux-gnu 
+            # fortran-cross-toolchain: gfortran-14-powerpc64le-linux-gnu
+            # cc: "powerpc64le-linux-gnu-gcc-9"
+            # cxx: "powerpc64le-linux-gnu-g++-9"
+            # fc: "powerpc64le-linux-gnu-gfortran-14"
+            ccache-max: 64M
+          - arch: s390x
+            target-triple: s390x-linux-gnu
+            # cross-toolchain: g++-9-s390x-linux-gnu 
+            # fortran-cross-toolchain: gfortran-14-s390x-linux-gnu
+            # cc: "s390x-linux-gnu-gcc-9"
+            # cxx: "s390x-linux-gnu-g++-9"
+            # fc: "s390x-linux-gnu-gfortran-14"
+            ccache-max: 64M
+          - arch: riscv64
+            target-triple: riscv64-linux-gnu
+            # cross-toolchain: g++-9-riscv64-linux-gnu 
+            # fortran-cross-toolchain: gfortran-14-riscv64-linux-gnu
+            # cc: "riscv64-linux-gnu-gcc-9"
+            # cxx: "riscv64-linux-gnu-g++-9"
+            # fc: "riscv64-linux-gnu-gfortran-14"
+            ccache-max: 64M
+
+    name: cross-compile (${{ matrix.arch }})
+
+    env:
+      CC: clang -target ${{ matrix.target-triple }}
+      CXX: clang++ -target ${{ matrix.target-triple }}
+      FC: ${{ matrix.target-triple }}-gfortran-14
```

Comment:
Using the current version of gfortran of the respective version of Ubuntu is probably fine.
``` suggestion
      FC: ${{ matrix.target-triple }}-gfortran
```